connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -145,6 +154,10 @@ public WorkerSinkTask(ConnectorTaskId id,
         this.isTopicTrackingEnabled = workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG);
         this.taskStopped = false;
         this.workerErrantRecordReporter = workerErrantRecordReporter;
+        Map<String, Object> adminProps = new HashMap<>();
+        adminProps.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, workerConfig.bootstrapServers());
```

Comment:
Connect supports overriding Kafka client configs per connector - see [KIP-458](https://cwiki.apache.org/confluence/display/KAFKA/KIP-458%3A+Connector+Client+Config+Override+Policy). If we're using an admin client for a specific connector (or task), it should respect these overrides - the connector may even be targeting a Kafka cluster different from the Connect cluster's backing Kafka cluster and we cannot use worker configurations to create Kafka clients for connectors. It would probably be cleaner to instantiate this admin client in the `Worker` and pass it to the `WorkerSinkTask` during creation. An example from `WorkerSourceTask` creation that might help - https://github.com/apache/kafka/blob/454b72161a76b1687a1263157d7cc30a1bdb2506/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L1424-L1446

Reply:
Ok. Thanks for pointing this out. Will make the changes as suggested.

Reply:
I added the logic to create Admin client in Worker and pass it along to WorkerSinkTask.